ci(suite-web): reorganize build and deploy tasks

prusnak opened the pull request:
Fixes https://github.com/trezor/trezor-suite/issues/2416

This PR reworks build and deploy tasks for `landing-page` and `suite-web`.

After the change, there are three builds targets and two deploy targets:

`build dev` - builds with "dev" settings
`build beta` - builds with "beta" settings
`build stable` - builds with "stable" settings

`deploy staging-wallet` - deploys beta build to staging-wallet.trezor.io
`deploy staging-suite` - deploys stable build to staging-suite.trezor.io (the deploy key is not yet in the CI)

The third deploy target will be deployed later:
`deploy suite` - deploys stable build to suite.trezor.io

keraf commented on ci/packages/landing-page.yml:
```diff
@@ -13,32 +13,62 @@ landing-page build:
+  only:
+    <<: *auto_run_branches
+  script:
+    - assetPrefix=/ @trezor/landing-page build
```
`assetPrefix` is not needed here and it's missing `yarn workspace`

keraf reviewed (changes requested):
Found an issue (see comment). I guess there's also update to the s3deploy script and documentation for production deployments? Or you're going to do that one in a separate PR?

prusnak commented:
Ah, right. I'll update the s3sync scripts later today in this PR.

prusnak commented:
Addressed the comments in fixup commit 294a4f2d891a88bfc3bf54632b3a143390269a97

prusnak commented:
This somehow breaks `suite-web e2e` tests - help how to fix this welcome.

mroz22 commented:
I will look what's is wrong with tests, It appears that tests do not see the page at all, possibly something with deployment or asset prefix or something like this.

prusnak commented:
Let's merge https://github.com/trezor/trezor-suite/pull/2471 (once it is done). Then I'll update this PR to fix the assetPrefix and deploy scripts for `suite-landing-page`. Changing this to Draft until #2471 is merged in.

prusnak commented:
I rebased the branch on top of the current develop and added the necessary changes for the newly added `suite-web-landing`.

The goal of this PR is to have fully functional deploy to staging-wallet (beta environment) and staging-suite (production environment).

The following manual steps in the CI should not fail and should successfully deploy to staging environments:

beta:
* `landing-page deploy staging-wallet`
* `suite-web deploy staging-wallet`

prod:
* `suite-web-landing deploy staging-suite`
* `suite-web deploy staging-suite`

CCing @tsusanka as we spoke today about the staging deployment from the CI and I want to keep him in the loop

prusnak commented:
We still need to address the e2e failures before this is merged into develop

mroz22 commented:
Dev deployment should go here?  https://suite.corp.sldev.cz/suite-web/new-paths 
I am asking because I have just noticed that it does not work - button "log in" takes you to 404.

Tests try to hit /wallet and won't find anything there https://gitlab.com/satoshilabs/trezor/trezor-suite/-/jobs/773827206/artifacts/file/packages/integration-tests/projects/suite-web/videos/backup/fail-t2.test.ts.mp4

But staging seems to work. So I suspect wrong configuration of dev servers, presumably related to this https://github.com/trezor/trezor-suite/issues/2485

prusnak commented:
I missed two changes which now I have added to 535f753698590da5fc157b3d2caf2d43e37478ec - let's see if that helps.

prusnak commented:
So it seems the change in https://github.com/trezor/trezor-suite/commit/535f753698590da5fc157b3d2caf2d43e37478ec did the trick, thanks @mroz22 for pointing me into the right direction.

I rebased the PR, squashed the fixup commit. Can you @keraf and @mroz22 do the final review before we merge this to develop?

szymonlesisz commented on ci/packages/suite-web.yml:
```diff
@@ -17,65 +17,60 @@ variables:
+  script:
+    - mkdir -p ${DEPLOY_DIRECTORY}
+    - cd packages/suite-web
+    - rsync --delete -va build/ "${DEPLOY_DIRECTORY}/"
```
i think this should be preserved, otherwise new landing page is not available in sldev build (i'm seeing old one)
```
- cd ../landing-page-web	
- rsync -abviuzP -va build/ "${DEPLOY_BASE_DIR}/suite-web/${CI_BUILD_REF_NAME}/"
```
